airbyte-webapp/src/components/GeographyDropdown/GeographyDropdown.tsx

```diff
@@ -0,0 +1,110 @@
+import { faPlus } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import React, { useState } from "react";
+import { useIntl } from "react-intl";
+import { components, OptionProps, MenuListProps, StylesConfig } from "react-select";
+
+import "/node_modules/flag-icons/css/flag-icons.min.css";
+
+import { DropDown } from "components/ui/DropDown";
+import { DropdownProps } from "components/ui/DropDown";
+
+import { Geography } from "core/request/AirbyteClient";
+import { theme } from "packages/cloud/theme";
+
+import styles from "./GeographyDropdown.module.scss";
+
+const GeographyOption: React.FC<OptionProps> = (props) => {
+  const { formatMessage } = useIntl();
+  return (
+    <components.Option className={styles.option} {...props}>
+      <div className={styles.flag}>
+        <span className={`fi fi-${props.label === "auto" ? "us" : props.label}`} />
+      </div>
+      <span className={styles.label}>{formatMessage({ id: `geography.${props.label}` })}</span>
+    </components.Option>
+  );
+};
+
+const MenuList = (props: MenuListProps) => {
+  const { formatMessage } = useIntl();
+  return (
+    <components.MenuList {...props}>
+      {props.children}
+      <div className={styles.menuList}>
+        <FontAwesomeIcon icon={faPlus} />
+        <span>{formatMessage({ id: "geography.new" })}</span>
+      </div>
+    </components.MenuList>
+  );
+};
+
+interface GeographySelectOption {
+  label: Geography;
+  value: Geography;
+}
+
+export const GeographyDropdown: React.FC<DropdownProps<GeographySelectOption>> = ({ options }) => {
+  const [option, setOption] = useState({});
+  const { formatMessage } = useIntl();
+  const handleOptionSelect = (event: { label: string }) => {
+    setOption(event.label);
+  };
+
+  const formatOptionLabel = ({ label }: { label: string }) => {
+    return (
+      <div className={styles.optionLabel}>
+        <div className={styles.flag}>
+          <span className={`fi fi-${label === "auto" ? "us" : label}`} />
+        </div>
+        <span className={styles.label}>
+          {formatMessage({ id: `geography.${label}`, defaultMessage: label.toUpperCase() })}
+        </span>
+      </div>
+    );
+  };
+
+  const customStyles: StylesConfig<GeographySelectOption> = {
+    option: (base, state) => {
+      let backgroundColor = theme.white;
```

Comment:
The `theme` files are going to go away, since they are only used for our legacy styled-components system. We should not use them in new code anymore. Instead if you need to get access in TS to colors, you should explicitally export them from your own SCSS file, e.g. see https://github.com/airbytehq/airbyte/blob/master/airbyte-webapp/src/components/ui/StatusIcon/CircleLoader.module.scss#L21-L25

Ideally though whenever possible you should prefer setting `className` instead and have all styling in the SCSS file (but there are legit cases when that's not necessarily possible, I'm not sure if this place here might support `className` actually?).

Reply:
I advised to import from `scss/export.module.scss` [here](https://github.com/airbytehq/airbyte/pull/18543#discussion_r1010322891), but seems like re-exporting is the preferred method. Sorry for the bad advice @natalyjazzviolin!

FWIW I think you could do all this styling in `GeographyDropdown.module.scss`. You can beat `react-select`'s CSS specificity by using something like `:global(.reactSelect__option--is-focused).option { ... }`. Note that `:global()` is special [CSS modules syntax](https://github.com/css-modules/css-modules#exceptions) that lets you reference a non-scoped "normal" CSS class.

Reply:
@josephkmh `react-select`'s specificity was trumping the `GeographyDropdown.module.scss` styles, but this is really helpful, thank you! I'll get right on it!